[Doc] Update field (#2994)
taichi-dev/taichi: docs/lang/articles/basic/field.md

> `3 x 2` matrix for each of its entry, use the statement

Reviewer: Make lines 67-69 become
````
In a general case, suppose you have a `128 x 64` field called `A`, and each element is
a `3 x 2` matrix, you can define it with
````

`A = ti.Matrix.field(3, 2, dtype=ti.f32, shape=(128, 64))`.

### Access matrix field elements
- If you want to get the matrix of grid node `i, j`, please use
  `mat = A[i, j]`. `mat` is simply a `3 x 2` matrix.
- To get the element on the first row and second column of that
  matrix, use `mat[0, 1]` or `A[i, j][0, 1]`.

> - The 0-D matrix fields `x = ti.Matrix.field(n=3, m=4, dtype=ti.f32, shape=())` should be accessed by `x[None][p, q]`, (`0<= p < n, 0<= q < m`).

Reviewer: ```suggestion
- The 0-D matrix field `x = ti.Matrix.field(n=3, m=4, dtype=ti.f32, shape=())` should be accessed by `x[None][p, q]` (`0 <= p < n, 0 <= q < m`).
```

:::note
- As you may have noticed, there are **two** indexing operators `[]`

>   when you load a matrix element from a global matrix field: the

Reviewer: ```suggestion
  when you access a member of a matrix from a matrix field: the
```

>   first is for field indexing, the second for matrix indexing.

Reviewer: ```suggestion
  first is for field indexing, and the second is for matrix indexing.
```

# set the position of the first particle to origin
particle_field[0] # local ti.Struct
particle_field[0].pos = ti.Vector([0.0, 0.0, 0.0])

> particle_field[0].pos[0] = 1.0

Reviewer: ```suggestion
```
This doesn't match the comment in L123.

Author: Thanks for your review!
